src/string/RGB.ts

```diff
@@ -0,0 +1,161 @@
+/**
+ * Represents strings which are valid RGB colors. Permits both absolute and percentage based values.
+ *
+ * @since 0.0.4
+ */
+import { Kind, Kind2, URIS, URIS2, HKT2 } from 'fp-ts/HKT'
+import * as D from 'io-ts/Decoder'
+import * as Enc from 'io-ts/Encoder'
+import * as fc from 'fast-check'
+import * as Eq_ from 'fp-ts/Eq'
+import * as G from 'io-ts/Guard'
+import * as Str from 'fp-ts/string'
+import * as TD from 'io-ts/TaskDecoder'
+import * as t from 'io-ts/Type'
+import * as Arb from '../internal/ArbitraryBase'
+import { pipe } from 'fp-ts/function'
+
+/**
+ * @since 0.0.4
+ * @category Internal
+ */
+interface RGBBrand {
+  readonly RGB: unique symbol
+}
+
+/**
+ * Represents strings which are valid RGB colors. Permits both absolute and percentage based values.
+ *
+ * @since 0.0.4
+ * @category Model
+ */
+export type RGB = string & RGBBrand
+
+/**
+ * @since 0.0.4
+ * @category Model
+ */
+export type SchemableParams<S> = HKT2<S, string, RGB>
+
+/**
+ * @since 0.0.4
+ * @category Model
+ */
+export type SchemableParams1<S extends URIS> = Kind<S, RGB>
+
+/**
+ * @since 0.0.4
+ * @category Model
+ */
+export type SchemableParams2<S extends URIS2> = Kind2<S, string, RGB>
+
+/**
+ * @since 0.0.4
+ * @category Model
+ */
+export type SchemableParams2C<S extends URIS2> = Kind2<S, unknown, RGB>
+
+/** @internal */
+const rgbColor =
+  /^rgb\((([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5]),){2}([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\)$/
+
+/** @internal */
+const rgbaColor =
+  /^rgba\((([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5]),){3}(0?\.\d|1(\.0)?|0(\.0)?)\)$/
+
+/** @internal */
+const rgbColorPercent = /^rgb\((([0-9]%|[1-9][0-9]%|100%),){2}([0-9]%|[1-9][0-9]%|100%)\)/
+
+/** @internal */
+const rgbaColorPercent =
+  /^rgba\((([0-9]%|[1-9][0-9]%|100%),){3}(0?\.\d|1(\.0)?|0(\.0)?)\)/
```

Comment:
since I was digging into the spec anyway, I also found that `rgb` and `rgba` function _identically_, with either able to take an optional alpha channel.

also also, there are some legal values that this doesn't cover. for instance:

- `rgb(255 255 255 / .5)` (commas not necessary, but if they're omitted then the optional alpha channel is separated with a slash)
- `rgb(none none 255 / none)` (sans commas, you can give "none" for any channel)
- `rgb(100% 100% 100% / 0.15)` (can also take percentages, and the alpha channel doesn't have to match. weirdly, the three color channels do all have to match each other.)

Reply:
These are good points!  I'll make a ticket for updating the module.  I was just copying from validators.js but it seems like they could stand some improvement

Reply:
Here is the issue: https://github.com/jacob-alford/schemable-ts-types/issues/92